packages/block-library/src/image/view.js

```diff
@@ -187,6 +201,58 @@ const { state, actions, callbacks } = store(
 					}
 				}
 			},
+			preloadImage() {
+				const { imageId } = getContext();
+				const imageMetadata = state.metadata[ imageId ];
+				const uploadedSrc = imageMetadata.uploadedSrc;
+
+				// Bails if it has already been preloaded. This could help
+				// prevent unnecessary preloading of the same image multiple times,
+				// leading to duplicate link elements in the document head.
+				if ( state.preloadedImageIds.has( imageId ) ) {
+					return;
+				}
+
```

Comment:
Nit: This can short-circuit earlier:

```suggestion
				// Bails if it has already been preloaded. This could help
				// prevent unnecessary preloading of the same image multiple times,
				// leading to duplicate link elements in the document head.
				if ( state.preloadedImageIds.has( imageId ) ) {
					return;
				}

				const imageMetadata = state.metadata[ imageId ];
				const uploadedSrc = imageMetadata.uploadedSrc;

```